src/script/lua_api/l_base.cpp

```diff
@@ -84,3 +85,15 @@ bool ModApiBase::registerFunction(lua_State *L, const char *name,
 
 	return true;
 }
+
+/**
+ * A number is Nan if rawequals of itself return false
+ * See Lua doc (https://www.lua.org/manual/5.3/manual.html)
+ * @param L
+ * @param idx
+ * @return isNaN status
+ */
+bool ModApiBase::isNaN(lua_State *L, int idx)
+{
+	return lua_type(L, idx) == LUA_TNUMBER && std::isnan(lua_tonumber(L, idx));
```

Comment:
This doesn't work if MT is compiled with `-ffast-math`

Reply:
@rubenwardy can you give me the source article ?

Reply:
https://stackoverflow.com/questions/22931147/stdisinf-does-not-work-with-ffast-math-how-to-check-for-infinity

I don't think there's any way around this. Is there a lua specific way of checking whether a float is NaN or the result of a divide by zero?

Reply:
O3 doesn't add fast-math. It seems to be a non standard and dangerous flag to use which should be enabled for some types of programs working on finite only.

https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

Reply:
Maybe it's possible to use an `attribute` to disable Ofast compiler optimisation at this point.
Something like `__attribute__((optimize (O3)))`?